packages/app/src/cli/services/dev/processes/dev-session/dev-session.ts

```diff
@@ -77,6 +100,53 @@ export class DevSession {
     await this.logger.debug(
       `✅ Event handled [Network: ${endHRTimeInMs(networkStartTime)}ms - Total: ${endHRTimeInMs(event.startTime)}ms]`,
     )
+
+    if (this.appEventsQueue.length > 0) {
+      await this.processEvents()
```

Comment:
I think we should handle this with a loop, not recursion. Because the function owns the boolean status, if we nest things, it might get to be incorrect.